src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php

```diff
@@ -62,7 +62,7 @@ public function process(BaseOrderInterface $order)
             return;
         }
 
-        $newPayment = $order->getLastNewPayment();
+        $newPayment = $order->getLastCartPayment();
```

Comment:
The logic is not good here... after the order is placed, we should no longer create payments in `cart` state, this needs to be well-thought